### PR TITLE
[zh-CN] decastdeljau

### PR DESCRIPTION
**Pull request by josherich**

### CODE REVIEW

File: components/sections/decasteljau/content.zh-CN.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+- 把相邻标记连成线，得到 `n-2` 根线。
+
+- 标记，连线，这样重复下去。
+
+- 直到剩下一根线。这根线上 `t` 的位置就是曲线 `t` 的位置。
+
+<div className="howtocode">
+
+### 如何实现 `de Casteljau` 算法
```
Comment (Pomax) [inline components/sections/decasteljau/content.zh-CN.md]: If there is no additional markup in the title, then there should be no additional markup in this heading either.

File: components/sections/decasteljau/content.zh-CN.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
+# de Casteljau 算法
+
+要计算贝塞尔曲线，我们必须遍历 `t` 的所有值，从 `0` 到 `1`，计算权重函数，得到 `x/y` 的值。当曲线很复杂时，计算量变得非常巨大。利用 `de Casteljau` 算法（一种几何画法），你可以轻易地用笔和尺画出曲线。
```
Comment (Pomax) [inline components/sections/decasteljau/content.zh-CN.md]: You're using backticks in a lot of places, but backticks are certainly not part of standard Chinese orthography. Please change all the strings that are currently marked up with backticks to use reference quoting instead (whatever the convention for that is in mainland Chinese).

**Comment by Pomax** [comment]
@thrillerist would you be able to review this PR?

File: components/sections/decasteljau/content.zh-CN.md [hunk 3]
```diff
@@ -1,28 +1,28 @@
 # de Casteljau 算法
 
-要计算贝塞尔曲线，我们必须遍历 `t` 的所有值，从 `0` 到 `1`，计算权重函数，得到 `x/y` 的值。当曲线很复杂时，计算量变得非常巨大。利用 `de Casteljau` 算法（一种几何画法），你可以轻易地用笔和尺画出曲线。
+要计算贝塞尔曲线，我们必须遍历 `t` 的所有值，从 0 到 1，计算权重函数，得到 `x/y` 的值。当曲线很复杂时，计算量变得非常巨大。利用 “de Casteljau” 算法（一种几何画法），你可以轻易地用笔和尺画出曲线。
```
Comment (Pomax) [inline components/sections/decasteljau/content.zh-CN.md]: Are there commonly spaces in texts like these? I know that in Japanese, there aren't, so if you're adding here "for clarity" then please take them back out and we should file an issue to make sure that CSS takes care of this for us.

**Comment by Pomax** [comment]
(file on the repo: https://github.com/Pomax/BezierInfo-2/blob/master/components/sections/explanation/content.zh-CN.md)

**Comment by josherich** [comment]
I would recommend using the "for styling" spaces in Chinese, looks more elegant, anyway, I'll remove them for now.

**Comment by Pomax** [comment]
"looks" come from styling, not from spaces in content. If extra space is needed to make things look better, them that can be achieved as part of either the build process, or through good use of CSS (possible with JS support). I've filed https://github.com/Pomax/BezierInfo-2/issues/88 and the issue of "making it look better" can be discussed there.

**Comment by josherich** [comment]
thanks, agree, styling spaces if necessary should be part of build process

**Comment by Pomax** [comment]
Thanks for the changes! hopefully @thrillerist has time to give your PR a quick look over, if not I'll probably merge it in sometime tomorrow.

File: components/sections/decasteljau/content.zh-CN.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+      y = (1-t) * points[i].y + t * points[i+1].y
+      newpoints[i] = new point(x,y)
+    drawCurve(newpoints, t)
+```
+
+以上算法做了什么? 如果参数 `points` 只有一个点， 就画出一个点。如果有多个点，就生成以 `t` 为比例的一系列点（例如，以上算法中的 `markers`），然后用新的点列表画出图像。
+
+</div>
+
+在以下的图表中，用鼠标移动计算 `de Casteljau` 算法的基础点， 左右移动鼠标，可以看到曲线是如何生成的。
```
Comment (thrillerist) [inline components/sections/decasteljau/content.zh-CN.md]: 我们通过实际操作观察这个过程。在以下的图表中，移动鼠标来改变用de Casteljau算法计算得到的曲线点， 左右移动鼠标，可以看到这种方法是如何生成曲线的。

File: components/sections/decasteljau/content.zh-CN.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+<Graphic preset="simple"title="Traversing a curve using de Casteljau's algorithm" setup={this.setup} draw={this.draw}/>
```
Comment (thrillerist) [inline components/sections/decasteljau/content.zh-CN.md]: <Graphic preset="simple"title="用de Casteljau算法来遍历曲线" setup={this.setup} draw={this.draw}/>

File: components/sections/decasteljau/content.zh-CN.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
+# de Casteljau 算法
+
+要计算贝塞尔曲线，我们必须遍历 `t` 的所有值，从 `0` 到 `1`，计算权重函数，得到 `x/y` 的值。当曲线很复杂时，计算量变得非常巨大。利用 `de Casteljau` 算法（一种几何画法），你可以轻易地用笔和尺画出曲线。
+
+我们用以下算法来替代微积分算法：
```
Comment (thrillerist) [inline components/sections/decasteljau/content.zh-CN.md]: 我们进行以下操作，而不是用微积分方程来找到某个`t`对应的`x/y`值：

**Comment by josherich** [comment]
add some changes from review, thanks for the review

**Comment by Pomax** [comment]
So I had a Chinese speaker read this over and they're saying that this file seems to be translated in a way that does not read naturally to a Chinese speaker. Much as it pains me to say I may have to close this PR and rely on @thrillerist to keep translating, as his translations are said to be quite natural Chinese.

**Comment by josherich** [comment]
yeah you can close it, I understand, it's fine

**Comment by Pomax** [comment]
Thanks. And sorry about this.

**Comment by hite** [comment]
@Pomax I think @josherich 's translations are naturally to me who live in China right now.
He does good works that I try to do. (I used to fork [BezierInfo-2-ZH_CN](https://github.com/hite/BezierInfo-2-ZH_CN). 

So I suggest @josherich fork it too.

PS: I  remember I made a pull request to `BezierInfo-2 ` repository  but I cann't find it anymore. 囧

**Pull request closed without merging**